enhancements/network/enable-flows-collection.md

```diff
@@ -0,0 +1,352 @@
+emission-collection traffic to the local node.
+
+According to the [architectural description of the flows' emitter/collectors pipeline](
+https://docs.google.com/document/d/1kzNfTLXhMolu8VTcH0m9hmvFluN9yKMy7TSNIXJVTrA/edit#heading=h.kulf936ct32p),
+the collector will be deployed as a DaemonSet, to ensure that there will be an
+instance of the collector for each node.
+
+Exposing the collector endpoint as a Kubernetes ClusterIP Service would cause the traffic
+to be distributed across all the nodes (unless you enable the `ServiceInternalTrafficPolicy`
+cluster-wide feature gate, which at this moment is in alpha status, only available
+for the latest Kubernetes 1.21 version).
+
+To limit the flows' traffic to internal nodes, each collector should expose a node
+port, so each OVN instance only has to know the node IP and Port (which might be
+fixed or discovered).
+
+From the user-side configuration of the IPFix cache and sampling, we should use a `ConfigMap` and
+expose it to OVN-Kubernetes with read permissions.
+
+Given the aforementioned details, following subsections depicts the work to do,
+grouped in different tasks.
+
+#### Task 1: modify OVN-Kubernetes configuration to discover automatically the collector address
+
+[Current `Network` operator configuration allows setting a static host:port address](
+https://docs.okd.io/latest/networking/ovn_kubernetes_network_provider/tracking-network-flows.html#nw-network-flows-object_tracking-network-flows).
+This information is forwarded to each OVN instance, on each cluster node. For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  exportNetworkFlows:
+    collectors:
+      - 192.168.1.99:2056
+```
+
+However, we need to set this address dynamically to ensure that each pod will forward
+the flows to the collector in its same node.
+
+We need to extend the previous configuration to allow replacing the `collectors` property
+by one of the following properties:
+
+* `nodePort: (int)`: if set, the OVN will discover the node IP and submit the flows to the
+  port indicated by such property. E.g. `nodePort: 30023`.
+* `selector: (map)`: if set, the OVN will search for any `Service` with `type: NodePort`
+  matching the labels passed as argument, then extract the service's `NodePort` and use
+  it alongside with the node IP. Configuration example:
+  ```yaml
+  selector:
+    role: flows-collector
+  ```
+
+#### Task 2: implement new discovery options in OVN-Kubernetes
+
+Before [OVN-Kubernetes invokes OpenVSwitch to enable flows](https://github.com/ovn-org/ovn-kubernetes/commit/ecc4047f5d0e732c267fccb2eabd64b5894b9f9a#diff-9c3b7372332fd16d60dba539a94dd0f819d262d1fbabd1e5d9c09e47da5af0ffR119-R128),
+if the network provider is configured for discovery, it needs to fetch the host/port and pass it
+to OpenVSwitch, in the `targets=` argument.
+
+If, by any reason, the Collector host:port can't be discovered (e.g. because the collector is not
```

Comment:
I don't think this is something that should be implemented in core OVN-K meaning  the health of the collector service is not something OVN-K should fetch and have to report

Reply:
I see your point. Probably we could create another component to work as a health checker.